lib/PerlIO.pm

```diff
@@ -288,14 +347,22 @@ DOS-like platforms and depending on the setting of C<$ENV{PERLIO}>:
  # [1] "stdio" if Configure found out how to do "fast stdio" (depends
  # on the stdio implementation) and in Perl 5.8, otherwise "unix perlio"
 
+=head2 Querying the layers of filehandles
+
+The following returns the B<names> of the PerlIO layers on a filehandle.
+
+   my @layers = PerlIO::get_layers($fh); # Or FH, *FH, "FH".
+
+The layers are returned in the order an open() or binmode() call would
+use them, and without colons.
+
 By default the layers from the input side of the filehandle are
 returned; to get the output side, use the optional C<output> argument:
 
    my @layers = PerlIO::get_layers($fh, output => 1);
 
 (Usually the layers are identical on either side of a filehandle but
-for example with sockets there may be differences, or if you have
-been using the C<open> pragma.)
+for example with sockets there may be differences.)
```

Comment:
To clarify this change: the open pragma (or more specifically ${^OPEN}) does not cause any differences here. Whether you set it for read or write handles changes which handles it applies to, but it applies to both directions of any handle it applies to.